File: bindings/lua/Polycode/SceneParticleEmitter.lua
require "Polycode/SceneMesh"

class "SceneParticleEmitter" (SceneMesh)

SceneParticleEmitter.PARTICLE_TYPE_POINT = 0
SceneParticleEmitter.PARTICLE_TYPE_QUAD = 1
SceneParticleEmitter.PARTICLE_TYPE_MESH = 2

function SceneParticleEmitter:__getvar(name)
	if name == "useScaleCurve" then
		return Polycode.SceneParticleEmitter_get_useScaleCurve(self.__ptr)
	elseif name == "scaleCurve" then
		local retVal = Polycode.SceneParticleEmitter_get_scaleCurve(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["BezierCurve"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "useColorCurves" then
		return Polycode.SceneParticleEmitter_get_useColorCurves(self.__ptr)
	elseif name == "colorCurveR" then
		local retVal = Polycode.SceneParticleEmitter_get_colorCurveR(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["BezierCurve"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "colorCurveG" then
		local retVal = Polycode.SceneParticleEmitter_get_colorCurveG(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["BezierCurve"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "colorCurveB" then
		local retVal = Polycode.SceneParticleEmitter_get_colorCurveB(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["BezierCurve"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "colorCurveA" then
		local retVal = Polycode.SceneParticleEmitter_get_colorCurveA(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["BezierCurve"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "colorDeviation" then
		local retVal = Polycode.SceneParticleEmitter_get_colorDeviation(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
	if SceneMesh["__getvar"] ~= nil then
		return SceneMesh.__getvar(self, name)
	end
end

function SceneParticleEmitter:__setvar(name,value)
	if name == "useScaleCurve" then
		Polycode.SceneParticleEmitter_set_useScaleCurve(self.__ptr, value)
		return true
	elseif name == "scaleCurve" then
		Polycode.SceneParticleEmitter_set_scaleCurve(self.__ptr, value.__ptr)
		return true
	elseif name == "useColorCurves" then
		Polycode.SceneParticleEmitter_set_useColorCurves(self.__ptr, value)
		return true
	elseif name == "colorCurveR" then
		Polycode.SceneParticleEmitter_set_colorCurveR(self.__ptr, value.__ptr)
		return true
	elseif name == "colorCurveG" then
		Polycode.SceneParticleEmitter_set_colorCurveG(self.__ptr, value.__ptr)
		return true
	elseif name == "colorCurveB" then
		Polycode.SceneParticleEmitter_set_colorCurveB(self.__ptr, value.__ptr)
		return true
	elseif name == "colorCurveA" then
		Polycode.SceneParticleEmitter_set_colorCurveA(self.__ptr, value.__ptr)
		return true
	elseif name == "colorDeviation" then
		Polycode.SceneParticleEmitter_set_colorDeviation(self.__ptr, value.__ptr)
		return true
	end
	if SceneMesh["__setvar"] ~= nil then
		return SceneMesh.__setvar(self, name, value)
	else
		return false
	end
end
function SceneParticleEmitter:SceneParticleEmitter(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "SceneMesh" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SceneParticleEmitter(unpack(arg))
	end
end

function SceneParticleEmitter:setParticleCount(newParticleCount)
	local retVal = Polycode.SceneParticleEmitter_setParticleCount(self.__ptr, newParticleCount)
end

function SceneParticleEmitter:getParticleCount()
	local retVal =  Polycode.SceneParticleEmitter_getParticleCount(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setParticleLifetime(lifetime)
	local retVal = Polycode.SceneParticleEmitter_setParticleLifetime(self.__ptr, lifetime)
end

function SceneParticleEmitter:getParticleLifetime()
	local retVal =  Polycode.SceneParticleEmitter_getParticleLifetime(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setDirectionDeviation(newDeviation)
	local retVal = Polycode.SceneParticleEmitter_setDirectionDeviation(self.__ptr, newDeviation.__ptr)
end

function SceneParticleEmitter:getDirectionDeviation()
	local retVal =  Polycode.SceneParticleEmitter_getDirectionDeviation(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:setEmitterSize(newSize)
	local retVal = Polycode.SceneParticleEmitter_setEmitterSize(self.__ptr, newSize.__ptr)
end

function SceneParticleEmitter:getEmitterSize()
	local retVal =  Polycode.SceneParticleEmitter_getEmitterSize(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:setGravity(newGravity)
	local retVal = Polycode.SceneParticleEmitter_setGravity(self.__ptr, newGravity.__ptr)
end

function SceneParticleEmitter:getGravity()
	local retVal =  Polycode.SceneParticleEmitter_getGravity(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:fixedUpdate()
	local retVal =  Polycode.SceneParticleEmitter_fixedUpdate(self.__ptr)
end

function SceneParticleEmitter:updateParticles()
	local retVal =  Polycode.SceneParticleEmitter_updateParticles(self.__ptr)
end

function SceneParticleEmitter:triggerParticles(allAtOnce)
	local retVal = Polycode.SceneParticleEmitter_triggerParticles(self.__ptr, allAtOnce)
end

function SceneParticleEmitter:enableParticleSystem(val)
	local retVal = Polycode.SceneParticleEmitter_enableParticleSystem(self.__ptr, val)
end

function SceneParticleEmitter:setUseFloorPlane(val)
	local retVal = Polycode.SceneParticleEmitter_setUseFloorPlane(self.__ptr, val)
end

function SceneParticleEmitter:setFloorPlaneOffset(floorPlaneOffset)
	local retVal = Polycode.SceneParticleEmitter_setFloorPlaneOffset(self.__ptr, floorPlaneOffset)
end

function SceneParticleEmitter:setFloorDamping(floorDamping)
	local retVal = Polycode.SceneParticleEmitter_setFloorDamping(self.__ptr, floorDamping)
end

function SceneParticleEmitter:setParticlesInWorldSpace(val)
	local retVal = Polycode.SceneParticleEmitter_setParticlesInWorldSpace(self.__ptr, val)
end

function SceneParticleEmitter:getParticlesInWorldSpace()
	local retVal =  Polycode.SceneParticleEmitter_getParticlesInWorldSpace(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setPerlinEnabled(val)
	local retVal = Polycode.SceneParticleEmitter_setPerlinEnabled(self.__ptr, val)
end

function SceneParticleEmitter:getPerlinEnabled()
	local retVal =  Polycode.SceneParticleEmitter_getPerlinEnabled(self.__ptr)
	return retVal
end

function SceneParticleEmitter:getParticleSpeed()
	local retVal =  Polycode.SceneParticleEmitter_getParticleSpeed(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setParticleSpeed(speed)
	local retVal = Polycode.SceneParticleEmitter_setParticleSpeed(self.__ptr, speed)
end

function SceneParticleEmitter:setPerlinValue(perlinValue)
	local retVal = Polycode.SceneParticleEmitter_setPerlinValue(self.__ptr, perlinValue.__ptr)
end

function SceneParticleEmitter:getPerlinValue()
	local retVal =  Polycode.SceneParticleEmitter_getPerlinValue(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:setParticleType(particleType)
	local retVal = Polycode.SceneParticleEmitter_setParticleType(self.__ptr, particleType)
end

function SceneParticleEmitter:getParticleType()
	local retVal =  Polycode.SceneParticleEmitter_getParticleType(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setParticleSize(particleSize)
	local retVal = Polycode.SceneParticleEmitter_setParticleSize(self.__ptr, particleSize)
end

function SceneParticleEmitter:getParticleSize()
	local retVal =  Polycode.SceneParticleEmitter_getParticleSize(self.__ptr)
	return retVal
end

function SceneParticleEmitter:setParticleRotationSpeed(rotationSpeed)
	local retVal = Polycode.SceneParticleEmitter_setParticleRotationSpeed(self.__ptr, rotationSpeed.__ptr)
end

function SceneParticleEmitter:getParticleRotationSpeed()
	local retVal =  Polycode.SceneParticleEmitter_getParticleRotationSpeed(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:setParticleDirection(direction)
	local retVal = Polycode.SceneParticleEmitter_setParticleDirection(self.__ptr, direction.__ptr)
end

function SceneParticleEmitter:getParticleDirection()
	local retVal =  Polycode.SceneParticleEmitter_getParticleDirection(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneParticleEmitter:setLoopParticles(val)
	local retVal = Polycode.SceneParticleEmitter_setLoopParticles(self.__ptr, val)
end

function SceneParticleEmitter:getLoopParticles()
	local retVal =  Polycode.SceneParticleEmitter_getLoopParticles(self.__ptr)
	return retVal
end

function SceneParticleEmitter:getNumSourceMeshes()
	local retVal =  Polycode.SceneParticleEmitter_getNumSourceMeshes(self.__ptr)
	return retVal
end

function SceneParticleEmitter:removeSourceMeshAtIndex(index)
	local retVal = Polycode.SceneParticleEmitter_removeSourceMeshAtIndex(self.__ptr, index)
end

function SceneParticleEmitter:positionParticle(index)
	local retVal = Polycode.SceneParticleEmitter_positionParticle(self.__ptr, index)
end

function SceneParticleEmitter:__delete()
	if self then Polycode.delete_SceneParticleEmitter(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneParticleEmitter.js
require('Polycode/SceneMesh')

function SceneParticleEmitter(particleCount,lifetime,speed) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneParticleEmitter(particleCount,lifetime,speed)
	}
	Object.defineProperties(this, {
		'useScaleCurve': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_useScaleCurve, set: SceneParticleEmitter.prototype.__set_useScaleCurve},
		'scaleCurve': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_scaleCurve, set: SceneParticleEmitter.prototype.__set_scaleCurve},
		'useColorCurves': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_useColorCurves, set: SceneParticleEmitter.prototype.__set_useColorCurves},
		'colorCurveR': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_colorCurveR, set: SceneParticleEmitter.prototype.__set_colorCurveR},
		'colorCurveG': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_colorCurveG, set: SceneParticleEmitter.prototype.__set_colorCurveG},
		'colorCurveB': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_colorCurveB, set: SceneParticleEmitter.prototype.__set_colorCurveB},
		'colorCurveA': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_colorCurveA, set: SceneParticleEmitter.prototype.__set_colorCurveA},
		'colorDeviation': { enumerable: true, configurable: true, get: SceneParticleEmitter.prototype.__get_colorDeviation, set: SceneParticleEmitter.prototype.__set_colorDeviation}
	})
}

SceneParticleEmitter.prototype = Object.create(SceneMesh.prototype);

SceneParticleEmitter.prototype.__get_useScaleCurve = function() {
	return Polycode.SceneParticleEmitter__get_useScaleCurve(this.__ptr)
}

SceneParticleEmitter.prototype.__set_useScaleCurve = function(val) {
	Polycode.SceneParticleEmitter__set_useScaleCurve(this.__ptr, val)
}

SceneParticleEmitter.prototype.__get_scaleCurve = function() {
	var retVal = new BezierCurve()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_scaleCurve(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_scaleCurve = function(val) {
	Polycode.SceneParticleEmitter__set_scaleCurve(this.__ptr, val.__ptr)
}

SceneParticleEmitter.prototype.__get_useColorCurves = function() {
	return Polycode.SceneParticleEmitter__get_useColorCurves(this.__ptr)
}

SceneParticleEmitter.prototype.__set_useColorCurves = function(val) {
	Polycode.SceneParticleEmitter__set_useColorCurves(this.__ptr, val)
}

SceneParticleEmitter.prototype.__get_colorCurveR = function() {
	var retVal = new BezierCurve()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_colorCurveR(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_colorCurveR = function(val) {
	Polycode.SceneParticleEmitter__set_colorCurveR(this.__ptr, val.__ptr)
}

SceneParticleEmitter.prototype.__get_colorCurveG = function() {
	var retVal = new BezierCurve()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_colorCurveG(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_colorCurveG = function(val) {
	Polycode.SceneParticleEmitter__set_colorCurveG(this.__ptr, val.__ptr)
}

SceneParticleEmitter.prototype.__get_colorCurveB = function() {
	var retVal = new BezierCurve()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_colorCurveB(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_colorCurveB = function(val) {
	Polycode.SceneParticleEmitter__set_colorCurveB(this.__ptr, val.__ptr)
}

SceneParticleEmitter.prototype.__get_colorCurveA = function() {
	var retVal = new BezierCurve()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_colorCurveA(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_colorCurveA = function(val) {
	Polycode.SceneParticleEmitter__set_colorCurveA(this.__ptr, val.__ptr)
}

SceneParticleEmitter.prototype.__get_colorDeviation = function() {
	var retVal = new Color()
	retVal.__ptr = 	Polycode.SceneParticleEmitter__get_colorDeviation(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.__set_colorDeviation = function(val) {
	Polycode.SceneParticleEmitter__set_colorDeviation(this.__ptr, val.__ptr)
}


SceneParticleEmitter.prototype.setParticleCount = function(newParticleCount) {
	Polycode.SceneParticleEmitter_setParticleCount(this.__ptr, newParticleCount)
}

SceneParticleEmitter.prototype.getParticleCount = function() {
	return Polycode.SceneParticleEmitter_getParticleCount(this.__ptr)
}

SceneParticleEmitter.prototype.setParticleLifetime = function(lifetime) {
	Polycode.SceneParticleEmitter_setParticleLifetime(this.__ptr, lifetime)
}

SceneParticleEmitter.prototype.getParticleLifetime = function() {
	return Polycode.SceneParticleEmitter_getParticleLifetime(this.__ptr)
}

SceneParticleEmitter.prototype.setDirectionDeviation = function(newDeviation) {
	Polycode.SceneParticleEmitter_setDirectionDeviation(this.__ptr, newDeviation)
}

SceneParticleEmitter.prototype.getDirectionDeviation = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getDirectionDeviation(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.setEmitterSize = function(newSize) {
	Polycode.SceneParticleEmitter_setEmitterSize(this.__ptr, newSize)
}

SceneParticleEmitter.prototype.getEmitterSize = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getEmitterSize(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.setGravity = function(newGravity) {
	Polycode.SceneParticleEmitter_setGravity(this.__ptr, newGravity)
}

SceneParticleEmitter.prototype.getGravity = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getGravity(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.fixedUpdate = function() {
	Polycode.SceneParticleEmitter_fixedUpdate(this.__ptr)
}

SceneParticleEmitter.prototype.Render = function(buffer) {
	Polycode.SceneParticleEmitter_Render(this.__ptr, buffer.__ptr)
}

SceneParticleEmitter.prototype.updateParticles = function() {
	Polycode.SceneParticleEmitter_updateParticles(this.__ptr)
}

SceneParticleEmitter.prototype.rebuildParticles = function(buffer) {
	Polycode.SceneParticleEmitter_rebuildParticles(this.__ptr, buffer.__ptr)
}

SceneParticleEmitter.prototype.triggerParticles = function(allAtOnce) {
	Polycode.SceneParticleEmitter_triggerParticles(this.__ptr, allAtOnce)
}

SceneParticleEmitter.prototype.enableParticleSystem = function(val) {
	Polycode.SceneParticleEmitter_enableParticleSystem(this.__ptr, val)
}

SceneParticleEmitter.prototype.setUseFloorPlane = function(val) {
	Polycode.SceneParticleEmitter_setUseFloorPlane(this.__ptr, val)
}

SceneParticleEmitter.prototype.setFloorPlaneOffset = function(floorPlaneOffset) {
	Polycode.SceneParticleEmitter_setFloorPlaneOffset(this.__ptr, floorPlaneOffset)
}

SceneParticleEmitter.prototype.setFloorDamping = function(floorDamping) {
	Polycode.SceneParticleEmitter_setFloorDamping(this.__ptr, floorDamping)
}

SceneParticleEmitter.prototype.setParticlesInWorldSpace = function(val) {
	Polycode.SceneParticleEmitter_setParticlesInWorldSpace(this.__ptr, val)
}

SceneParticleEmitter.prototype.getParticlesInWorldSpace = function() {
	return Polycode.SceneParticleEmitter_getParticlesInWorldSpace(this.__ptr)
}

SceneParticleEmitter.prototype.setPerlinEnabled = function(val) {
	Polycode.SceneParticleEmitter_setPerlinEnabled(this.__ptr, val)
}

SceneParticleEmitter.prototype.getPerlinEnabled = function() {
	return Polycode.SceneParticleEmitter_getPerlinEnabled(this.__ptr)
}

SceneParticleEmitter.prototype.getParticleSpeed = function() {
	return Polycode.SceneParticleEmitter_getParticleSpeed(this.__ptr)
}

SceneParticleEmitter.prototype.setParticleSpeed = function(speed) {
	Polycode.SceneParticleEmitter_setParticleSpeed(this.__ptr, speed)
}

SceneParticleEmitter.prototype.setPerlinValue = function(perlinValue) {
	Polycode.SceneParticleEmitter_setPerlinValue(this.__ptr, perlinValue)
}

SceneParticleEmitter.prototype.getPerlinValue = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getPerlinValue(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.setParticleType = function(particleType) {
	Polycode.SceneParticleEmitter_setParticleType(this.__ptr, particleType)
}

SceneParticleEmitter.prototype.getParticleType = function() {
	return Polycode.SceneParticleEmitter_getParticleType(this.__ptr)
}

SceneParticleEmitter.prototype.setParticleSize = function(particleSize) {
	Polycode.SceneParticleEmitter_setParticleSize(this.__ptr, particleSize)
}

SceneParticleEmitter.prototype.getParticleSize = function() {
	return Polycode.SceneParticleEmitter_getParticleSize(this.__ptr)
}

SceneParticleEmitter.prototype.setParticleRotationSpeed = function(rotationSpeed) {
	Polycode.SceneParticleEmitter_setParticleRotationSpeed(this.__ptr, rotationSpeed)
}

SceneParticleEmitter.prototype.getParticleRotationSpeed = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getParticleRotationSpeed(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.setParticleDirection = function(direction) {
	Polycode.SceneParticleEmitter_setParticleDirection(this.__ptr, direction)
}

SceneParticleEmitter.prototype.getParticleDirection = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SceneParticleEmitter_getParticleDirection(this.__ptr)
	return retVal
}

SceneParticleEmitter.prototype.setLoopParticles = function(val) {
	Polycode.SceneParticleEmitter_setLoopParticles(this.__ptr, val)
}

SceneParticleEmitter.prototype.getLoopParticles = function() {
	return Polycode.SceneParticleEmitter_getLoopParticles(this.__ptr)
}

SceneParticleEmitter.prototype.addSourceMesh = function(mesh) {
	Polycode.SceneParticleEmitter_addSourceMesh(this.__ptr, mesh.__ptr)
}

SceneParticleEmitter.prototype.getNumSourceMeshes = function() {
	return Polycode.SceneParticleEmitter_getNumSourceMeshes(this.__ptr)
}

SceneParticleEmitter.prototype.getSourcesMeshAtIndex = function(index) {
	var retVal = new Mesh()
	retVal.__ptr = Polycode.SceneParticleEmitter_getSourcesMeshAtIndex(this.__ptr, index)
	return retVal
}

SceneParticleEmitter.prototype.removeSourceMeshAtIndex = function(index) {
	Polycode.SceneParticleEmitter_removeSourceMeshAtIndex(this.__ptr, index)
}

SceneParticleEmitter.prototype.positionParticle = function(index) {
	Polycode.SceneParticleEmitter_positionParticle(this.__ptr, index)
}

SceneParticleEmitter.prototype.Clone = function(deepClone,ignoreEditorOnly) {
	var retVal = new Entity()
	retVal.__ptr = Polycode.SceneParticleEmitter_Clone(this.__ptr, deepClone, ignoreEditorOnly)
	return retVal
}

SceneParticleEmitter.prototype.applyClone = function(clone,deepClone,ignoreEditorOnly) {
	Polycode.SceneParticleEmitter_applyClone(this.__ptr, clone.__ptr, deepClone, ignoreEditorOnly)
}
